docs/proposals/DataTable/README.md

```diff
@@ -0,0 +1,165 @@
+# DataTable
+
+The `DataTable` Component would give the users options for pagination, sorting,
+and column visibility by leveraging some sort of API or service.
+
+## Design Patterns
+
+Tables are used to organize tabular information to help users understand and
+quickly scan for information.
+
+Some scenarios for Tables include the following:
+
+- Index Pages
+- Dashboards
+
+### Sorting
+
+Sorting can be enabled on column headers to improve usability.
+
+The default sorted column should be based on contextual based on user needs.
+
+To indicate the column being sorted, use an up (ascending) or down (descending)
+arrow next to the column title. Only 1 column can be sorted at a time. User can
+click on a sorted header to reverse the sorting order. Arrow will rotate 180
+degrees.
+
+Not all columns will need sorting enabled.
+
+# Column Visibility
+
+Not 100% sure if this is something to be implemented at component level but at
+least should be facilitated somehow.
+
+## Usage
+
+```tsx
+<DataTable
+  data={[
+    { name: "Gui", awesome: "yes" },
+    { name: "Harry", awesome: "yes" },
+    { name: "Chris", awesome: "yes" },
+    { name: "Juan", awesome: "yes" },
+  ]}
+  columns={[
+    {
+      accessorKey: "name",
+      cell: info => <span>{info.getValue()}</span>,
+    },
+    {
+      accessorKey: "awesome",
+      cell: info => info.getValue(),
+    },
+  ]}
+/>
+```
+
+## Interface and Props
+
+```ts
+interface DataTableProps<T> {
+  /**
+   * The actual data that will be used for the table.
+   * Typescript should infer T from typeof data.
+   *
+   * @type {T[]}
+   * @memberof DataTable
+   */
+  data: T[];
+  /**
+   * Should follow the @tanstack/react-table ColumnDef.
+   * I'm not sure if we need to create an abstraction around this
+   * specific type, ColumnDef by itself is already very flexible and powerfull enough
+   * to enable us to configure the table any way we like.
+   * https://tanstack.com/table/v8/docs/guide/column-defs
+   *
+   * @type {ColumnDef<T>[]}
+   * @memberof DataTable
+   */
+  columns: ColumnDef<T>[];
+
+  /**
+   * Enables pagination, mostly follows:
+   * https://tanstack.com/table/v8/docs/api/features/pagination
+   *
+   * @type {Pagination}
+   */
+  pagination?: Pagination;
```

Comment:
Without this prop we just display all data I assume?

Reply:
Yes, if no prop was passed in we will not show the pagination.

Reply:
That's the intention. 

```ts
pagination == undefined // no pagination
pagination != undefined // built-in or manually implemented pagination
```

and `manualPagination` is required.

Reply:
cool, thank you!